lib/ts/recipe/session/with-jwt/recipeImplementation.ts

```diff
@@ -0,0 +1,174 @@
+                        then the final payload will use the values they provide
+                    */
+                    sub: userId,
+                    ...accessTokenPayload,
+                },
+                validitySeconds: getJWTExpiry(accessTokenValidityInSeconds),
+            });
+
+            if (jwtResponse.status === "UNSUPPORTED_ALGORITHM_ERROR") {
+                // Should never come here
+                throw new Error("JWT Signing algorithm not supported");
+            }
+
+            accessTokenPayload = {
+                ...accessTokenPayload,
+                [config.jwtKey]: jwtResponse.jwt,
+            };
+
+            let sessionContainer = await originalImplementation.createNewSession({
+                res,
+                userId,
+                accessTokenPayload,
+                sessionData,
+            });
+
+            return new SessionClassWithJWT(sessionContainer, jwtRecipeImplementation, config);
+        },
+        getSession: async function ({
+            req,
+            res,
+            options,
+        }: {
+            req: any;
+            res: any;
+            options?: VerifySessionOptions;
+        }): Promise<SessionContainerInterface | undefined> {
+            let sessionContainer = await originalImplementation.getSession({ req, res, options });
+
+            if (sessionContainer === undefined) {
+                return undefined;
+            }
+
+            return new SessionClassWithJWT(sessionContainer, jwtRecipeImplementation, config);
+        },
+        refreshSession: async function ({ req, res }: { req: any; res: any }): Promise<SessionContainerInterface> {
+            let accessTokenValidityInSeconds = Math.ceil((await this.getAccessTokenLifeTimeMS()) / 1000);
+
+            // Refresh session first because this will create a new access token
+            let newSession = await originalImplementation.refreshSession({ req, res });
+            let accessTokenPayload = newSession.getAccessTokenPayload();
+
+            // Remove the old jwt
+            delete accessTokenPayload[config.jwtKey];
+
+            let jwtResponse = await jwtRecipeImplementation.createJWT({
+                payload: accessTokenPayload,
+                validitySeconds: getJWTExpiry(accessTokenValidityInSeconds),
+            });
+
+            if (jwtResponse.status === "UNSUPPORTED_ALGORITHM_ERROR") {
+                // Should never come here
+                throw new Error("JWT Signing algorithm not supported");
+            }
+
+            accessTokenPayload = {
+                ...accessTokenPayload,
+                [config.jwtKey]: jwtResponse.jwt,
+            };
+
+            await newSession.updateAccessTokenPayload(accessTokenPayload);
+
+            return new SessionClassWithJWT(newSession, jwtRecipeImplementation, config);
+        },
+        updateAccessTokenPayload: async function ({
```

Comment:
Have a test case where you:
- create a new session with some jwt payload
- Call this `updateAccessTokenPayload` function with that session's handle with a new payload
- Let the session refresh
- Check that the refreshed session's return value has the new payload and not the older one.

Reply:
Done